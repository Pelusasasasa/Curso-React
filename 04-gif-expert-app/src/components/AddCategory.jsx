import PropTypes from 'prop-types';
import {useState} from 'react';
const apikey = '2pn5YpKp47DQywZy4NpidWL0sg9Yq4IB';

export const AddCategory = ({onNewCategory,resetCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onSubmit = (event) => {

        const newValue = inputValue.trim();

        event.preventDefault();
        if (newValue.length <= 1) return;
        
        onNewCategory(newValue);
        setInputValue('');
    }

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    };

    const onReset = () => {
        resetCategory();
    }

  return (
    <div className='form'>
        <form aria-label='form' onSubmit={ onSubmit }>
            <input 
                type="text"
                placeholder="Buscar gifs"
                value = {inputValue}
                onChange = { onInputChange }
            />
        </form>

        <button onClick={onReset}>Reset</button>
    </div>
  )
};


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}