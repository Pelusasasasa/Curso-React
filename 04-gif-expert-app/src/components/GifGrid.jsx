import { getGifs } from "../helpres/getGifs"

export const GifGrid = ({category}) => {
    getGifs(category);

    return (
        <>
            <h3>{category}</h3>
            <p>hola Mundo</p>
        </>
  )

}