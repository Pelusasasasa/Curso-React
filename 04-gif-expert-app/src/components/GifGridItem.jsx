import PropTypes from 'prop-types';

export const GifGridItem = ({id,title,url}) => {
  return (
    <div className="card" key={id}>
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
};


GifGridItem.protoType = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}