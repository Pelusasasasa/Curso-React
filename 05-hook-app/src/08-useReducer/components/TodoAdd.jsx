import { useState } from 'react'
import { useForm } from '../../hooks/useForm';

export const TodoAdd = ({onNewTodo}) => {
  
  const {description,onInputChange,resetForm} = useForm({
    description: ''
  });

  const onFormSubmit = (e) => {
    e.preventDefault();
    if (value !== "") {
      onNewTodo({
        id: new Date().getTime(),
        description: value,
        done: false,
      })
    }
  }

  return (
    <form onSubmit={onFormSubmit}>
        <input type="text"
         name='description'
         value={description}
         onChange={onInputChange} placeholder="Â¿Que hay que hacer?" className="form-control"/>
        <button type="submit" className="btn btn-outline-primary mt-1">Agregar</button>
    </form>
  )
}
