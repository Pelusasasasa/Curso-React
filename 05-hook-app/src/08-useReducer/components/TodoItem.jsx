export const TodoItem = ({id,description, onDeleteTodo}) => {


  return (
    <li id={id} className="list-group-item d-flex justify-content-between">
        <span className="aling-self-center">{description}</span>
        <button onClick={() => onDeleteTodo(id)} className="btn btn-danger">Borrar</button>
    </li>
  )
}
