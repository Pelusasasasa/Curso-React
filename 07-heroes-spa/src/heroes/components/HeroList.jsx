import React from 'react'
import { getHerosByPublisher } from '../helpers'

export const HeroList = ({publisher}) => {

    const heroes = getHerosByPublisher(publisher);

  return (
    <ul>
        {heroes.map( hero => (
            <li key={heroes.id}>{hero.superhero}</li>
        ))}
    </ul>
  )
}
