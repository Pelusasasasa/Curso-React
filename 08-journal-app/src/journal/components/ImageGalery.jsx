import { ImageListItem, ImageList } from '@mui/material';

export const ImageGalery = ({ images = [] }) => {
  return (
    <ImageList sx={{ width: '100%', height: 500 }} cols={4} rowHeight={164}>
      {images.map((item) => (
        <ImageListItem key={item}>
          <img
            srcSet={`${item}?w=164&h=164&fit=crop&auto=format&dpr=2 2x`}
            src={`${item}?w=164&h=164&fit=crop&auto=format`}
            alt='Imagen de Galeria'
            loading="lazy"
          />
        </ImageListItem>
      ))}
    </ImageList>
  );
}