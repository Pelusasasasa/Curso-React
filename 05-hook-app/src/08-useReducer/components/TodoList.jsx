import {TodoItem} from './TodoItem';

export const TodoList = ({todos, onDeleteTodo}) => {
  return (
    <ul className='list-group'>
        {
            todos.map( todo => (
                <TodoItem 
                 key={todo.id}
                 id={todo.id} 
                 done={todo.done} 
                 description={todo.description} 
                 onDeleteTodo={ onDeleteTodo }
                 />
            ))
        }
    </ul>
  )
}
