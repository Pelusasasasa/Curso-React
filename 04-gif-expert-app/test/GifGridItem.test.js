import { GifGridItem } from "../src/components/GifGridItem";

describe('Probando el grif grid item', () => {

    test('Haciendo match con  el snapchot', () => { 

        const title = 'Saitama';
        const url = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX';

        const {container} = <GifGridItem title={title} url={url}/>

        expect(container).toMatchSnapshot();


     })


});