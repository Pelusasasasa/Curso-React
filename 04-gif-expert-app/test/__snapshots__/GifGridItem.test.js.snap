// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el grif grid item Haciendo match con  el snapchot 1`] = `undefined`;
